graphql_compiler/compiler/blocks.py

```diff
@@ -403,22 +420,30 @@ class OutputSource(BasicBlock):
     See the comment on the @output_source directive in schema.py on why this is necessary.
     """
 
-    def __init__(self):
-        """Create a new OutputSource block."""
-        super(OutputSource, self).__init__()
+    def validate():
+        """This type of block is always valid in isolation."""
+        pass
+
+
+class Fold(MarkerBlock):
+    """A marker for the start of a @fold context."""
+
+    def __init__(self, root_location):
+        """Create a new Fold block rooted at the given location."""
+        super(Fold, self).__init__()
+        self.root_location = root_location
         self.validate()
 
     def validate(self):
-        """Ensure that the OutputSource block is valid.
+        """Ensure the Fold block is valid."""
+        if not isinstance(self.root_location, Location):
+            raise ValueError(u'Expected a Location for root_location, got: '
+                             u'{}'.format(self.root_location))
```

Comment:
is this supposed to be `type(self.root_location)`?

Reply:
I'll add the type as well, but it's nice to know the concrete value for the sake of debugging.